plugins/html-elements-plugin/src/main/java/ru/sbtqa/tag/pagefactory/reflection/HtmlReflectionImpl.java

```diff
@@ -69,36 +88,71 @@
-                && isChildOf(HtmlElement.class, field);
+    @Override
+    public WebElement findElementInBlockByTitle(Page page, String blockPath, String title) throws PageException {
+        return findElementInBlockByTitle(page, blockPath, title, WebElement.class);
     }
 
-    /**
-     * Check whether given field is a child of specified class
-     *
-     * @param parent class that is supposed to be parent
-     * @param field field to check
-     * @return true|false
-     */
-    public static boolean isChildOf(Class<?> parent, Field field) {
-        Class<?> fieldType = field.getType();
-        while (fieldType != null && fieldType != Object.class) {
-            if (fieldType == parent) {
-                return true;
+    @Override
+    public <T extends WebElement> List<T> findListOfElements(Page page, String listTitle, Class<T> type) throws PageException {
+        return findListOfElements(listTitle, type, page);
+    }
+
+    @Override
+    public <T extends WebElement> List<T> findListOfElements(String listTitle, Class<T> type, Object context) throws PageException {
+        for (Field field : FieldUtilsExt.getDeclaredFieldsWithInheritance(context.getClass())) {
+            if (isRequiredElement(field, listTitle) && List.class.isAssignableFrom(field.getType())
+                    && field.getGenericType() instanceof ParameterizedType
+                    && ((ParameterizedType) field.getGenericType()).getActualTypeArguments()[0].equals(type)) {
+                field.setAccessible(true);
+                try {
+                    return (List<T>) field.get(context);
+                } catch (IllegalAccessException e) {
+                    throw new FactoryRuntimeException(
+                            String.format("Internal error during attempt to find list '%s'", listTitle), e);
+                }
             }
-            fieldType = fieldType.getSuperclass();
         }
+        throw new ElementNotFoundException(String.format("Couldn't find elements list '%s' on page '%s'", listTitle, PageContext.getCurrentPageTitle()));
+    }
 
-        return false;
+    @Override
+    public List<WebElement> findListOfElements(Page page, String listTitle) throws PageException {
+        return findListOfElements(page, listTitle, WebElement.class);
+    }
+
+    @Override
+    public <T extends WebElement> List<T> findListOfElementsInBlock(Page page, String blockPath, String listTitle, Class<T> type) throws PageException {
+        Object block = findBlock(page, blockPath);
+        return findListOfElements(listTitle, type, block);
+    }
+
+    @Override
+    public List<WebElement> findListOfElementsInBlock(Page page, String blockPath, String listTitle) throws PageException {
+        return findListOfElementsInBlock(page, blockPath, listTitle, WebElement.class);
+    }
+
+    @Override
+    public HtmlElement findBlock(Page page, String blockPath) throws NoSuchElementException {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove the declaration of thrown exception 'org.openqa.selenium.NoSuchElementException' which is a runtime exception. ([squid:RedundantThrowsDeclarationCheck](https://sonarcloud.io/organizations/sbtqasonarkey/rules#rule_key=squid%3ARedundantThrowsDeclarationCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=ru.sbtqa.tag%3Apage-factory-2&pullRequest=101&issues=AWbDPCupSgN4uCclZPgY&open=AWbDPCupSgN4uCclZPgY)

[ISSUE_KEY]: AWbDPCupSgN4uCclZPgY